Hexastore.Web/Queue/QueueContainer.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Hexastore.Web.EventHubs;
+using Microsoft.Extensions.Logging;
+
+namespace Hexastore.Web.Queue
+{
+    public class QueueContainer : IQueueContainer, IDisposable
+    {
+        private readonly EventReceiver _eventReceiver;
+        private readonly QueueWriter[] _queueWriters;
+        private readonly ILogger<QueueContainer> _logger;
+        private readonly int _count = 32;
+        private bool _running = true;
+
+        public QueueContainer(EventReceiver eventReceiver, ILogger<QueueContainer> logger)
+        {
+            _logger = logger;
+            _eventReceiver = eventReceiver;
+            _queueWriters = new QueueWriter[_count];
+            for(int i=0; i< _count; i++) {
+                _queueWriters[i] = new QueueWriter(eventReceiver, logger);
+            }
+            _ = _eventReceiver.LogCount();
+            _ = LogQueueLength();
+        }
+
+        public void Dispose()
+        {
+            _running = false;
+            foreach(var q in _queueWriters) {
+                q.Dispose();
+            }
+        }
+
+        public void Send(StoreEvent storeEvent)
+        {
+            var partitionId = Math.Abs(storeEvent.PartitionId.GetHashCode() % _count);
```

Comment:
This doesn't take into account that two events with the same partition could be in different stores. It feels like we are moving to lock that is faster than before, but still limited or one that requires the client to understand the entire flow of hexadb to optimize things themselves.

Should we consider a memory queue that feeds into a smarter multi-key lock instead of relying only on partitions?

Reply:
Discussed this offline - infinite parallelism does not really help us because there is a limit to the parallelism from a pure hardware perspective. Here we are trusting that the distribution is a fair one and the scheduler works well.